libsoftwarecontainer/src/gateway/networkgateway.cpp

```diff
@@ -38,172 +39,24 @@ NetworkGateway::~NetworkGateway()
 ReturnCode NetworkGateway::readConfigElement(const json_t *element)
 {
     IPTableEntry e;
-    std::string chain;
-    if (!read(element, "type", chain)) {
-        log_error() << "No type specified in network config.";
-        return ReturnCode::FAILURE;
-    }
-
-    if ("INCOMING" == chain) {
-        e.m_type = "INPUT";
-    } else if ("OUTGOING" == chain) {
-        e.m_type = "OUTPUT";
-    } else {
-        log_error() << e.m_type << " is not a valid type ('INCOMING' or 'OUTGOING')";
-        return ReturnCode::FAILURE;
-    }
-
-    int p;
-    if (!read(element, "priority", p)) {
-        log_error() << "No priority specified in network config.";
-        return ReturnCode::FAILURE;
-    }
-    e.m_priority = p;
-
-    if (e.m_priority < 1) {
-        log_error() << "Priority can not be less than 1 but is " << e.m_priority;
-        return ReturnCode::FAILURE;
-    }
-
-    const json_t *rules = json_object_get(element, "rules");
-
-    if (rules == nullptr) {
-        log_error() << "No rules specified";
-        return ReturnCode::FAILURE;
-    }
+    NetworkGatewayParser configParser;
+    ReturnCode returnValue = configParser.parseNetworkGatewayConfiguration(element, e);
 
-    if (!json_is_array(rules)) {
-        log_error() << "Rules not specified as an array";
-        return ReturnCode::FAILURE;
+    if (ReturnCode::SUCCESS == returnValue) {
+        m_entries.push_back(e);
     }
```

Comment:
yeah but actuall in case of error, parser says some blues sentences and the function returns error to parseConfig which also outputs error. So imho it would be redundant info.